src/main/java/com/backbase/ct/bbfuel/client/user/UserPresentationRestClient.java

```diff
@@ -46,4 +58,38 @@ public UserGetResponseBody getUserByExternalId(String userExternalId) {
             .as(UserGetResponseBody.class);
     }
 
+    public void createIdentityUserAndLogResponse(UsersPostRequestBody user){
+
+        Response response = createIdentity(user);
+
+
+        if (isBadRequestException(response, "User already exists")) {
+            log.info("User [{}] already exists, skipped ingesting this user", user.getExternalId());
+        } else if (response.statusCode() == SC_CREATED) {
+            log.info("Identity User [{}] ingested under legal entity [{}]",
+                user.getExternalId(), user.getLegalEntityExternalId());
+        } else {
+            response.then()
+                .statusCode(SC_CREATED);
+        }
+
+
+    }
+
+    private Response createIdentity(UsersPostRequestBody user){
+
+        IdentitiesPostRequestBody createUserBody = new IdentitiesPostRequestBody();
+
+        createUserBody
+            .withExternalId(user.getExternalId())
+            .withLegalEntityInternalId(legalEntityPresentationRestClient.retrieveLegalEntityByExternalId(user.getLegalEntityExternalId()).getId())
+            .withFullName(user.getFullName())
+            .withEmailAddress(user.getExternalId() + "@email.com");
```

Comment:
This could be a constant "@email.com"?

Reply:
That should be replaced. "email.com" is a valid Domain name and we don't want to send emails to a real mail server

Please use "example.com" or "email.invalid" as an alternative